pmd-java/src/main/java/net/sourceforge/pmd/lang/java/typeresolution/TypeHelper.java

```diff
@@ -24,28 +24,55 @@ private TypeHelper() {
      * @return <code>true</code> if type node n is of type clazzName or a subtype of clazzName
      */
     public static boolean isA(final TypeNode n, final String clazzName) {
+        final Class<?> clazz = loadClassWithNodeClassloader(n, clazzName);
+
+        if (clazz != null) {
+            return isA(n, clazz);
+        }
+
+        return clazzName.equals(n.getImage()) || clazzName.endsWith("." + n.getImage());
+    }
+    
+    /**
+     * Checks whether the resolved type of the given {@link TypeNode} n is exactly of the type
+     * given by the clazzName.
+     *
+     * @param n the type node to check
+     * @param clazzName the class name to compare to
+     * @return <code>true</code> if type node n is exactly of type clazzName.
+     */
+    public static boolean isExactlyA(final TypeNode n, final String clazzName) {
+        final Class<?> clazz = loadClassWithNodeClassloader(n, clazzName);
+
+        if (clazz != null) {
+            return n.getType() == clazz;
```

Comment:
For the isExactly test, wouldn't it be enough to compare the FQCN of `n.getType()` and the `clazzName`? Technically, we can't be wrong, and we'd avoid hitting the classloader

Reply:
What are you suggesting? Doing something in the lines of `n.getType().getCannonicalName()` and testing it against `clazzName`? I'm uncertain that will work in the cases `getCannonicalName` will return null (local and anonymous classes)....

Reply:
But the name is binary, meaning we would get stuff such as `foo.bar$baz` which is not a FQCN (what we ask the user as input). We would have to manually deal with strings to deal with that...

Reply:
Hum yeah you're probably right